GVFS/GVFS.Common/Http/GitEndPointResponseData.cs

```diff
@@ -59,6 +59,11 @@ public string RetryableReadToEnd()
                 throw new RetryableException("Stream is null (this could be a result of network flakiness), retrying.");
             }
 
+            if (!this.Stream.CanRead)
+            {
+                throw new RetryableException("Stream is not readable (this could be a result of network flakiness), retrying.");

```

Comment:
Will we still be able to find this in telemetry (to understand how frequently it's happening)?

Reply:
From current telemetry we know it's happening a very small percentage of the time.

The RetryableExceptions currently don't write to telemetry.  Funny you mention it cause I was talking to @mjcheetham about doing a separate PR to add telemetry on these.  I have an inkling this.Stream == null no longer happens since the dispose issues were previously fixed and adding that telemetry will help with that story as well.